File: cpp/28_number_spiral_diagonals.cpp
#include <iostream>

int number_spiral_diagonals(){
    int total = 1;
    // n:            3,  5,  7,  9, ...
    // n^2:          9, 25, 49, 81, ...
    // n^2 -  n + 1: 7, 21, 43, 73, ...
    // n^2 - 2n + 2: 5, 17, 37, 65, ...
    // n^2 - 3n + 3: 3, 13, 31, 57, ...
    // total of corners for each n: 4n^2 - 6n + 6
    for (int n = 3; n <= 1001; n += 2){
        total += 4*n*n - 6*n + 6;
    }
    return total;
}

int main() {
    std::cout << number_spiral_diagonals() << "\n";
}

File: c#/_28_NumberSpiralDiagonals.cs
using System;

public class _28_NumberSpiralDiagonals{
	public static int Euler28(){
		int sum = 1;
		// n = 3: corners 9, 7, 5, 3
		// n = 5: corners 25, 21, 17, 13
		// n = 7: corners 49, 43, 37, 31
		// n = 9: corners 81, 73, 65, 57
		// n = x: corners x^2, x^2 - x + 1, x^2 - 2x + 2, x^2 - 3x + 3
		// sum of corners: 4x^2 - 6x + 6
		for (int n = 3; n <= 1001; n += 2){
			sum += 4 * n * n - 6 * n + 6;
		}
		return sum;
	}
	
	public static void Main(){
		Console.WriteLine(Euler28());
	}
}
